src/DynamicExpresso.Core/Parsing/Parser.cs

```diff
@@ -1443,6 +1441,22 @@ private bool TryParseKnownType(string name, out Type type)
 			var originalPos = _token.pos;
 			_arguments.TryGetKnownType(name, out type);
 
+			type = ParseKnownGenericType(name, type);
```

Comment:
This change is purely cosmetic: I've introduced a new method `ParseKnownGenericType` to make the code a bit easier to understand.